config/timesyncd.go

```diff
@@ -0,0 +1,175 @@
+package config
+
+import (
+	"fmt"
+	"github.com/godbus/dbus/v5"
+	"github.com/godbus/dbus/v5/introspect"
+	"github.com/godbus/dbus/v5/prop"
+	"github.com/home-assistant/os-agent/utils/lineinfile"
+	"regexp"
+	"strings"
+
+	logging "github.com/home-assistant/os-agent/utils/log"
+)
+
+const (
+	objectPath    = "/io/hass/os/Config/Timesyncd"
+	ifaceName     = "io.hass.os.Config.Timesyncd"
+	timesyncdConf = "/etc/systemd/timesyncd.conf"
+)
+
+var (
+	optNTPServer         []string
+	optFallbackNTPServer []string
+	configFile           = lineinfile.LineInFile{FilePath: timesyncdConf}
+)
+
+type timesyncd struct {
+	conn  *dbus.Conn
+	props *prop.Properties
+}
+
+func getNTPServers() []string {
+	return getTimesyncdConfigProperty("NTP")
+}
+
+func getFallbackNTPServers() []string {
+	return getTimesyncdConfigProperty("FallbackNTP")
+}
+
+func setNTPServer(c *prop.Change) *dbus.Error {
+	servers, ok := c.Value.([]string)
+	if !ok {
+		return dbus.MakeFailedError(fmt.Errorf("invalid type for NTPServer"))
+	}
+
+	value := strings.Join(servers, " ")
+
+	if err := setTimesyncdConfigProperty("NTP", value); err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	optNTPServer = servers
+	return nil
+}
+
+func setFallbackNTPServer(c *prop.Change) *dbus.Error {
+	servers, ok := c.Value.([]string)
+	if !ok {
+		return dbus.MakeFailedError(fmt.Errorf("invalid type for FallbackNTPServer"))
+	}
+
+	value := strings.Join(servers, " ")
+
+	if err := setTimesyncdConfigProperty("FallbackNTP", value); err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	optFallbackNTPServer = servers
+	return nil
+}
+
+func getTimesyncdConfigProperty(property string) []string {
+	value, err := configFile.Find(`^\s*(`+property+`=).*$`, `\[Time\]`, true)
+
+	var servers []string
+
+	if err != nil || value == nil {
+		return servers
+	}
+
+	matches := regexp.MustCompile(property + `=([^\s#]+(?:\s+[^\s#]+)*)`).FindStringSubmatch(*value)
+	if len(matches) > 1 {
+		servers = strings.Split(matches[1], " ")
+	}
+
+	return servers
+}
+
+func setTimesyncdConfigProperty(property string, value string) error {
+	var params = lineinfile.NewPresentParams("NTP=" + value)
+	params.Regexp, _ = regexp.Compile(`^\s*#?\s*(` + property + `=).*$`)
+	// Keep it simple, timesyncd.conf only has the [Time] section
+	params.After = `\[Time\]`
+	if err := configFile.Present(params); err != nil {
+		return fmt.Errorf("failed to set %s: %w", property, err)
+	}
+
+	if err := restartTimesyncd(); err != nil {
+		return fmt.Errorf("failed to restart timesyncd: %w", err)
+	}
```

Comment:
I think I'd prefer this to be a separate `Restart` method, this allows to make multiple changes at once. It is not really relevant in this case, but as we add support for more systemd configurations, we'll have the same pattern everywhere.